VACMS-2261_3602: Add resources panel to clp & create downloadable external file type

ethanteague opened the pull request:
## Description
See #2261 #3602 

## Testing done
Visual / Behat

## Screenshots
![image](https://user-images.githubusercontent.com/2404547/100887025-7ea3f800-3482-11eb-9186-ca259afb4636.png)
![image](https://user-images.githubusercontent.com/2404547/100887061-8a8fba00-3482-11eb-887c-9287b086dbb8.png)
![image](https://user-images.githubusercontent.com/2404547/100887091-911e3180-3482-11eb-949a-1db45946253e.png)
![image](https://user-images.githubusercontent.com/2404547/100887114-98453f80-3482-11eb-9ce8-e4bab1bee5ac.png)
![image](https://user-images.githubusercontent.com/2404547/100887143-9f6c4d80-3482-11eb-975b-061df61c18ad.png)


## QA steps
- As an administrator, go to `/media/add/document_external` and  add an external pdf file, setting the node to published.
- On save, visually verify that the file type is populated.
- As an administrator, go to `node/add/campaign_landing_page#edit-group-downloadable-resources`
- Visually verify that the resources panel appears as indicated by the screenshots.
- Verify that Media resources -> Select files allows you to select the external file you added.
- Verify enabling panel and not entering data in fields does not save and displays form error.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [x] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

kevwalsh commented:
@ethanteague I updated some ACs based on our conversation with PW POs this morning, plus a couple of minor details. 

i unchecked the ACs that i modified.

ethanteague commented:
> @ethanteague I updated some ACs based on our conversation with PW POs this morning, plus a couple of minor details.
> 
> i unchecked the ACs that i modified.

Updated, @kevwalsh

swirtSJW commented on config/sync/field.field.media.document_external.field_media_in_library.yml:
```diff
@@ -0,0 +1,23 @@
+entity_type: media
+bundle: document_external
+label: Reusable
+description: 'By default, documents you upload to the CMS can be reused by other VA.gov CMS editors. Uncheck this box if to prevent reuse by other CMS editors.'
```
```suggestion
description: 'By default, documents you upload to the CMS can be reused by other VA.gov CMS editors. Uncheck this box to prevent reuse by other CMS editors.'
```

ethanteague replied:
:+1:

swirtSJW commented on docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PdfCheck.php:
```diff
@@ -0,0 +1,25 @@
+ *
+ * @Constraint(
+ *   id = "PdfCheck",
+ *   label = @Translation("UPDF Check", context = "Validation"),
```
UPDF?

swirtSJW commented on tests/behat/drupal-spec-tool/content_model_media_fields.feature:
```diff
@@ -5,20 +5,25 @@ Feature: Content model: Media fields
-| Media type | Video | Video URL | field_media_video_embed_field | Video Embed | Required | 1 | Video Textfield | Translatable |
+  Scenario: Fields
+    Then exactly the following fields should exist for entity type media
+      | Type       | Bundle              | Field label                 | Machine name                     | Field type         | Required | Cardinality | Form widget                           | Translatable |
```
Are the indening changes directly from spectool?

ethanteague replied:
Nope - my bad

swirtSJW commented on tests/behat/drupal-spec-tool/views.feature:
```diff
@@ -107,6 +108,9 @@ Feature: Views
 | Files | Master | default | Master |
 | Files | Files overview | page_1 | Page |
 | Files | File usage | page_2 | Page |
+| File browsers | Block | block_1 | Block |
```
Would like to get non-generic names on these views.  block_1 and default are not very intuitive when accessing through code.

swirtSJW commented:
Pulling this local to resolve conflict and test

swirtSJW commented:
Mime type gts picked up automtically.
![image](https://user-images.githubusercontent.com/5752113/101140497-0e5fb880-35e1-11eb-8658-61223910f403.png)

swirtSJW commented:
Am running into this where the widget does not get set on config... we have run into this before.  will have to set it by hand.

swirtSJW commented:
@ethanteague  I resolved the conflicts on this but then I needed to force push to your branch and didn't have perms for that so I opened up https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3631

There are a couple issues with it though.   
 1. I can not get the widget for choosing the file to work despite changing  the widget
![image](https://user-images.githubusercontent.com/5752113/101149423-ebd39c80-35ec-11eb-9db1-37bd2c943f41.png)

 2   There are js issues related to hiding fields that prevent node saving. 
![image](https://user-images.githubusercontent.com/5752113/101149404-e5ddbb80-35ec-11eb-8c8f-9696668b8e1b.png)

3. The toggling logic is not surviving node save.  I narrowed it down to it is working on prod, but not on staging.  So it was either something I introduced with the tinkering I did on Kaises PR that merged.

I need sleep.

swirtSJW commented:
It is something about these two fields... in the toggle logic.   
![image](https://user-images.githubusercontent.com/5752113/101155284-e417f600-35f4-11eb-91fd-eed1cd324a9a.png)

If I toggle them on and put some text in them,   things save fine and the toggle logic persists too.
